src/jsdoc_transformer.ts

```diff
@@ -616,7 +616,20 @@ export function jsdocTransformer(
               // getOriginalNode(decl) is required because the type checker cannot type check
               // synthesized nodes.
               const typeStr = moduleTypeTranslator.typeToClosure(ts.getOriginalNode(decl));
-              localTags.push({tagName: 'type', type: typeStr});
+              // If @define is present then do not add @type, but instead verify that any type is
+              // correct. Specifying any type at all here may become an error later.
+              const defineTag = localTags.find(({tagName}) => tagName === 'define');
+              if (!defineTag) {
+                localTags.push({tagName: 'type', type: typeStr});
+              } else if (defineTag.type !== typeStr) {
+                if (defineTag.type !== undefined) {
+                  moduleTypeTranslator.error(
+                      varStmt,
+                      `incorrect type in @define: written Closure type was '${
+                          defineTag.type}' but TypeScript inferred '${typeStr}'`);
```

Comment:
nit: inferred is a bit misleading here, it could so be an explicitly declared type.

Reply:
No longer applies

Reply:
Changed to 'found'

Reply:
Deleted again.